### PR TITLE
FRDQ-2 - Create button component

### PR DESCRIPTION
**Pull request by Krysik**

### CODE REVIEW

File: src/assets/button-icon-disabled.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (hayuna) [inline src/assets/button-icon-disabled.svg:1]: These icons are similar. Can we use just one and change the fill in specific conditions?

File: src/components/atoms/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const Button = ({
+  children,
+  icon = undefined,
+  iconPos = EIconPosition.NONE,
+  type = 'fill'
+}: IButton) => {
+  const content = () => {
+    if (icon && iconPos === EIconPosition.NONE) {
+      throw Error('You need to specify icon position (iconPos prop)');
```
Comment (hayuna) [inline src/components/atoms/Button/Button.tsx:19]: Dont you think that "Crashing page" is that so good idea to notify user that he need to specify some props ?

File: src/assets/button-icon-disabled.svg (hunk 1 above)
Comment (Krysik) [reply src/assets/button-icon-disabled.svg:1]: @hayuna  Is that should be possible to change icon by a prop ?

Comment (hayuna) [reply src/assets/button-icon-disabled.svg:1]: It should be. Components should be reusable. What if we want to and another icon in different place? Should we create another component to this purpose? Or just change one props?

Comment (Krysik) [reply src/assets/button-icon-disabled.svg:1]: Im worried there is no "clean" way to do that or I just can't :p

Comment (hayuna) [reply src/assets/button-icon-disabled.svg:1]: @Programmers-Only-Group/frontend-team Could you help him to achieve this?

Comment (hayuna) [reply src/assets/button-icon-disabled.svg:1]: @Krysik https://stackoverflow.com/a/54521330

File: src/App.tsx [hunk 3]
```diff
@@ -1,8 +1,42 @@
 import React from 'react';
+import {ReactComponent as btnIcon} from 'assets/button-icon.svg';
+import Button from 'components/atoms/Button';
+import { EIconPosition } from 'components/atoms/Button/Button.model';
```
Comment (hayuna) [inline src/App.tsx:4]: In each case of use this button, should we import this enum? I see that enum are problematic

**Comment by hayuna** [comment]
@hendrysiak Please look at this

File: src/App.tsx (hunk 3 above)
Comment (hendrysiak) [reply src/App.tsx:4]: I think, if in specific case we have only 2-3 specific strings, we can set it in interface e.q:  
position: 'left' | 'right' | 'center';  
TS will do rest ;)

**Review by PiotrGrobelak — COMMENTED** [review]
Pretty button prototype :)  What do you think about switch statements as separate functions instead of if/else in styles? Imo this pattern below is more readable for me an example with "position" variable.

 const setIconPosition = (pos: EIconPosition) => {
  switch (pos) {
    case EIconPosition.LEFT:
      return '0 16px 0 0';
    case EIconPosition.RIGHT:
      return '0 0 0 16px';
    case EIconPosition.NONE:
      return '0';
  }
};

In styles: 

margin: ${({ iconPos: pos }) => setIconPosition(pos)}};

**Review by PiotrGrobelak — COMMENTED** [review]
Sorry. I made a mistake at the function name.  This switch statement set margin, not position :) It should be 'setIconMargin' or something similar. Of course, I don't know if this method will be good :)

**Comment by Krysik** [comment]
> Sorry. I made a mistake at the function name. This switch statement set margin, not position :) It should be 'setIconMargin' or something similar. Of course, I don't know if this method will be good :)

Oh, I just copied your solution :p

**Review by hayuna — APPROVED** [review]

**Pull request merged**